sound/soc/sof/amd/acp-ipc.c

```diff
@@ -177,11 +185,18 @@ irqreturn_t acp_sof_ipc_irq_thread(int irq, void *context)
 		/* set the done bit */
 		acp_dsp_ipc_dsp_done(sdev);
 		spin_unlock_irq(&sdev->ipc_lock);
-		ipc_irq = true;
+		return IRQ_HANDLED;
+	}
+
+	acp_mailbox_read(sdev, sdev->debug_box.offset, &status, sizeof(u32));
+	if ((status & SOF_IPC_PANIC_MAGIC_MASK) == SOF_IPC_PANIC_MAGIC) {
+		bool non_recoverable = true;
+
+		snd_sof_dsp_panic(sdev, sdev->dsp_oops_offset, non_recoverable);
+		return IRQ_HANDLED;
```

Comment:
it's also surprising that there are two calls to 'panic' reading from two different places. That's worthy of a comment if that again was intentional.

Reply:
Yes, 

>>Getting panic After FW_READY ipc from DSP.
The First message from DSP is FW_READY. In the  FW_READY we will get window information (dsp, host, debug, exception box all offsets and sizes are set that will keep offsets same for both DSP and HOST).


>>Getting panic before  FW_READY .

Here DSP and HOST sides have different mailbox offsets and also exception box offset was not set. there is only one common offset is DSP mailbox offset. So reading panic status from DSP box first four bytes and after that (XOOPS,PANIC,STACK ) info in DSP mailbox.